imap-proto/src/core.rs

```diff
@@ -81,29 +87,28 @@ named!(pub literal<&[u8]>, do_parse!(
 /// string = quoted / literal
 named!(pub string<&[u8]>, alt!(quoted | literal));
 
+/// string bytes as as utf8
+named!(pub string_utf8<&str>, map_res!(string, str::from_utf8));
+
 /// nstring = string / nil
 named!(pub nstring<Option<&[u8]>>, alt!(
-    map!(tag_s!("NIL"), |_| None) |
+    map!(nil, |_| None) |
     map!(string, |s| Some(s))
 ));
 
-named!(pub string_utf8<&str>, map!(string, |s| str::from_utf8(s).unwrap()));
-
-named!(pub nstring_utf8<Option<&str>>, map!(nstring, |val| val.map(|s| str::from_utf8(s).unwrap())));
+/// nstring bytes as utf8
```

Comment:
Please don't mix code movement or reformatting in the same commit as functional changes, this makes it much harder for me to review.